### ISSUE TITLE
[documentation] Support for 1.7.10 using custom lwgjl patches

### ISSUE THREAD

**Issue by mazunki** [Admicos/minecraft-wayland#62]
This issue is mostly just for documentation purposes, in case anyone runs into the same issue as I did.

So I'm playing Gregtech: New Horizons as of lately, which runs on Minecraft 1.7.10. It uses [a custom version of lwgjl](https://github.com/GTNewHorizons/lwjgl3ify), which effectively turned this patch (or the similar `USE=wayland-only` on Gentoo) on Wayland incompatible.

I had to use the bundled version of lwjgl, because of a missing glfwInitAllocator function glfwInitAllocator ([introduced in =glfw-3.4](https://github.com/glfw/glfw/commit/22b586b3d87c91581074cc2a47fe5eba6f10bf0d). I wonder if @eigenraven can confirm they're using glfw-3.4 in the bundled version.

In any regard, manually upgrading to [glfw-3.4](https://github.com/glfw/glfw/releases/tag/3.4), which I'm going to get merged on `::gentoo` (with `USE="wayland"`), using the system GLFW version just works (and is compatible with her lwgjl patches). Wayland native, no patches! :D

**Comment by eigenraven** [comment]
I use upstream LWJGL3 following Minecraft vanilla snapshots, currently at 3.3.2, will soon update to 3.3.3. <https://github.com/LWJGL/lwjgl3/releases/tag/3.3.2> seems to indicate it's glfw 3.3.8, but I know that the lwjgl teams puts its own patches on top of the upstream glfw here https://github.com/LWJGL-CI/glfw

**Comment by mazunki** [comment]
Interesting, they must've backported/cherrypicked that change, then? Also good to know lwjgl may eventually reach glfw-3.4 too. Is it true that lwjgl is just the java interface into glfw, or am I misunderstanding the relationship?

**Comment by eigenraven** [comment]
yes, lwjgl3 is mostly Java/JNI bindings to native libraries, with very little logic of its own

**Comment by mazunki** [comment]
Thanks :)

I have realized since the last time I commented here that Cleanroom runs under XWayland for me, even if [it seems like Wayland is supported by Cleanroom/lwjgl3ifyl](https://github.com/GTNewHorizons/lwjgl3ify/issues?q=wayland). I haven't dug too deep into this rabbit hole, but I'll soon have some time to dig up this nest.
